### PR TITLE
Revert socket.io version upgrade

### PR DESCRIPTION
**Pull request by mszabo-wikia**
Reverts socket.io version upgrade in https://github.com/Wikia/app/pull/13065 - there are issues with Polish wikis.

### CODE REVIEW

File: extensions/wikia/Chat2/js/controllers/controllers.js [hunk 1]
```diff
@@ -59,13 +59,13 @@ var NodeChatSocketWrapper = $.createClass(Observable, {
...
 			socket.on('connect', this.proxy(function () {
-				this.onConnect(socket);
+				this.onConnect(socket, ['xhr-polling']);
 			}, this));
 			socket.on('reconnecting', connectionFail);
 		});
 	},
 
-	onConnect: function (socket) {
+	onConnect: function (socket, transport) {
```
Comment (WikiaTech) [inline extensions/wikia/Chat2/js/controllers/controllers.js:68]: MAJOR Remove the unused function parameter "transport". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AUnusedFunctionArgument)

**Comment by WikiaTech** [comment]
SonarQube analysis reported 59 issues
* ![MAJOR][MAJOR] 49 major
* ![MINOR][MINOR] 2 minor
* ![INFO][INFO] 8 info

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [controllers.js#L31](https://github.com/Wikia/app/blob/b6ec48321a5ce615c3f8991af0d0b37f36c8cf5d/extensions/wikia/Chat2/js/controllers/controllers.js#L31): Remove this logging statement. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AS2228)
1. ![MAJOR][MAJOR] [controllers.js#L94](https://github.com/Wikia/app/blob/b6ec48321a5ce615c3f8991af0d0b37f36c8cf5d/extensions/wikia/Chat2/js/controllers/controllers.js#L94): Add a "default" clause to this "switch" statement. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3ASwitchWithoutDefault)
1. ![MAJOR][MAJOR] [controllers.js#L140](https://github.com/Wikia/app/blob/b6ec48321a5ce615c3f8991af0d0b37f36c8cf5d/extensions/wikia/Chat2/js/controllers/controllers.js#L140): Replace "!=" with "!==". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AEqEqEq)
1. ![MAJOR][MAJOR] [controllers.js#L140](https://github.com/Wikia/app/blob/b6ec48321a5ce615c3f8991af0d0b37f36c8cf5d/extensions/wikia/Chat2/js/controllers/controllers.js#L140): Replace "==" with "===". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AEqEqEq)
1. ![MAJOR][MAJOR] [controllers.js#L144](https://github.com/Wikia/app/blob/b6ec48321a5ce615c3f8991af0d0b37f36c8cf5d/extensions/wikia/Chat2/js/controllers/controllers.js#L144): Replace "==" with "===". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AEqEqEq)
1. ![MAJOR][MAJOR] [controllers.js#L176](https://github.com/Wikia/app/blob/b6ec48321a5ce615c3f8991af0d0b37f36c8cf5d/extensions/wikia/Chat2/js/controllers/controllers.js#L176): Replace "!=" with "!==". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AEqEqEq)
1. ![MAJOR][MAJOR] [controllers.js#L216](https://github.com/Wikia/app/blob/b6ec48321a5ce615c3f8991af0d0b37f36c8cf5d/extensions/wikia/Chat2/js/controllers/controllers.js#L216): Remove the unused function parameter "index". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AUnusedFunctionArgument)
1. ![MAJOR][MAJOR] [controllers.js#L231](https://github.com/Wikia/app/blob/b6ec48321a5ce615c3f8991af0d0b37f36c8cf5d/extensions/wikia/Chat2/js/controllers/controllers.js#L231): Restrict what this loop acts on by testing each property. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AForIn)
1. ![MAJOR][MAJOR] [controllers.js#L256](https://github.com/Wikia/app/blob/b6ec48321a5ce615c3f8991af0d0b37f36c8cf5d/extensions/wikia/Chat2/js/controllers/controllers.js#L256): Replace "==" with "===". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AEqEqEq)
1. ![MAJOR][MAJOR] [controllers.js#L273](https://github.com/Wikia/app/blob/b6ec48321a5ce615c3f8991af0d0b37f36c8cf5d/extensions/wikia/Chat2/js/controllers/controllers.js#L273): Refactor this code to not nest more than 3 if/for/while/switch/try statements. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3ANestedIfDepth)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by WikiaTech** [comment]
SonarQube analysis reported 59 issues
* ![MAJOR][MAJOR] 49 major
* ![MINOR][MINOR] 2 minor
* ![INFO][INFO] 8 info

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [controllers.js#L31](https://github.com/Wikia/app/blob/b6ec48321a5ce615c3f8991af0d0b37f36c8cf5d/extensions/wikia/Chat2/js/controllers/controllers.js#L31): Remove this logging statement. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AS2228)
1. ![MAJOR][MAJOR] [controllers.js#L94](https://github.com/Wikia/app/blob/b6ec48321a5ce615c3f8991af0d0b37f36c8cf5d/extensions/wikia/Chat2/js/controllers/controllers.js#L94): Add a "default" clause to this "switch" statement. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3ASwitchWithoutDefault)
1. ![MAJOR][MAJOR] [controllers.js#L140](https://github.com/Wikia/app/blob/b6ec48321a5ce615c3f8991af0d0b37f36c8cf5d/extensions/wikia/Chat2/js/controllers/controllers.js#L140): Replace "==" with "===". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AEqEqEq)
1. ![MAJOR][MAJOR] [controllers.js#L140](https://github.com/Wikia/app/blob/b6ec48321a5ce615c3f8991af0d0b37f36c8cf5d/extensions/wikia/Chat2/js/controllers/controllers.js#L140): Replace "!=" with "!==". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AEqEqEq)
1. ![MAJOR][MAJOR] [controllers.js#L144](https://github.com/Wikia/app/blob/b6ec48321a5ce615c3f8991af0d0b37f36c8cf5d/extensions/wikia/Chat2/js/controllers/controllers.js#L144): Replace "==" with "===". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AEqEqEq)
1. ![MAJOR][MAJOR] [controllers.js#L176](https://github.com/Wikia/app/blob/b6ec48321a5ce615c3f8991af0d0b37f36c8cf5d/extensions/wikia/Chat2/js/controllers/controllers.js#L176): Replace "!=" with "!==". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AEqEqEq)
1. ![MAJOR][MAJOR] [controllers.js#L216](https://github.com/Wikia/app/blob/b6ec48321a5ce615c3f8991af0d0b37f36c8cf5d/extensions/wikia/Chat2/js/controllers/controllers.js#L216): Remove the unused function parameter "index". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AUnusedFunctionArgument)
1. ![MAJOR][MAJOR] [controllers.js#L231](https://github.com/Wikia/app/blob/b6ec48321a5ce615c3f8991af0d0b37f36c8cf5d/extensions/wikia/Chat2/js/controllers/controllers.js#L231): Restrict what this loop acts on by testing each property. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AForIn)
1. ![MAJOR][MAJOR] [controllers.js#L256](https://github.com/Wikia/app/blob/b6ec48321a5ce615c3f8991af0d0b37f36c8cf5d/extensions/wikia/Chat2/js/controllers/controllers.js#L256): Replace "==" with "===". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AEqEqEq)
1. ![MAJOR][MAJOR] [controllers.js#L273](https://github.com/Wikia/app/blob/b6ec48321a5ce615c3f8991af0d0b37f36c8cf5d/extensions/wikia/Chat2/js/controllers/controllers.js#L273): Refactor this code to not nest more than 3 if/for/while/switch/try statements. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3ANestedIfDepth)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Review by macbre — APPROVED** [review]

**Comment by macbre** [comment]
Issues are solved by 283666e (see #13111)

**Pull request merged**